### PR TITLE
Update macOS setup for 2021 and Apple Silicon [skip ci]

### PR DESCRIPTION
**Pull request by cat5inthecradle**
Update SETUP.md with new macos instructions for the state of 3rd party packages in fall 2021, and for development on Apple Silicon.

- rearranged some steps
- became more opinionated about use of zsh and nvm
- added notes for Apple Silicon M1 systems
- kept Catalina notes for reference, can possibly be deleted
- applied some suggestions from [markdownlint](https://github.com/DavidAnson/markdownlint)

## Links

- @simonguest got this working, [his notes](https://docs.google.com/document/d/1fH51x4HR_eHb53IWQHeTTm_-HhB8w60_FLKMmwUx9hE/edit#heading=h.onq0ngv25yrc) helped me get it working on my MacBook Air (M1, 2020).

## Testing story

I'd love to see this followed on one or more fresh M1 and Intel systems to confirm it is correct before merging.

## Deployment strategy

N/A

## Follow-up work

N/A

## Privacy

N/A

## Security

N/A

## Caching

N/A

## PR Checklist:

- [ ] ~Tests provide adequate coverage~
- [ ] ~Privacy and Security impacts have been assessed~
- [ ] Code is well-commented
- [ ] ~New features are translatable or updates will not break translations~
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -36,26 +36,88 @@ You can do Code.org development using OSX, Ubuntu, or Windows (running Ubuntu in
...
+
+1. Open your Terminal. These steps assume you are using **zsh**, the default for OSX.
+2. Optionally configure your **zsh** experience.
+   1. Either install [oh-my-zsh](https://ohmyz.sh/) and add the [git-prompt](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/git-prompt) plugin, or install the plugin directly without oh-my-zsh as described under the [OS X Catalina](#os-x-catalina) instructions.
+3. Install/Update **Xcode Command Line Tools** via `xcode-select --install`
+4. Install [Homebrew](https://brew.sh/), a macOS package manager
+5. Install **Redis** via `brew install redis`
+6. Install **MySql 5.7** via `brew install mysql@5.7`
+7. Install the **Java 8 JSK**
+   1. Either explicitly via `brew cask install adoptopenjdk/openjdk/adoptopenjdk8`
```
Comment (hannahbergam) [inline SETUP.md]: This errored for me- I had to run `brew install --cask adoptopenjdk/openjdk/adoptopenjdk8` instead

Comment (hannahbergam) [reply SETUP.md]: Actually, didn't work at all without Rosetta (M1)

File: SETUP.md [hunk 2]
```diff
@@ -36,26 +36,88 @@ You can do Code.org development using OSX, Ubuntu, or Windows (running Ubuntu in
...
+   2. Or by installing [sdkman](https://sdkman.io/) and installing a suitable JDK. Similar to **rbenv** and **nvm**, **sdkman** allows you to switch between versions of Java.
+      1. Different versions will be available depending on your system architecture, use `sdk list java` to identify a Java 8 JDK available for ARM architecture.
+      3. `sdk default java <installed version>` to ensure it is the default for future shells.
+8. Install **rbenv**
+   1. For non M1 systems, `rbenv install 2.5.0` should be sufficient
+   2. For Apple Silicon, special configuration is required to set *libffi* options correctly. The following is a single line to execute.
+      ```sh
+      export optflags="-Wno-error=implicit-function-declaration"; export LDFLAGS="-L/opt/homebrew/opt/libffi/lib"; export CPPFLAGS="-I/opt/homebrew/opt/libffi/include"; export PKG_CONFIG_PATH="/opt/homebrew/opt/libffi/lib/pkgconfig" | rbenv install 2.5.0
```
Comment (hannahbergam) [inline SETUP.md:121]: I wasn't able to get this to run, but I did get it working with `brew install rbenv` and then `rbenv install 2.5.0`

File: SETUP.md [hunk 3]
```diff
@@ -36,26 +36,88 @@ You can do Code.org development using OSX, Ubuntu, or Windows (running Ubuntu in
...
+   ```
+   curl -O https://raw.githubusercontent.com/zph/homebrew-cervezas/master/pdftk.rb
+   brew install ./pdftk.rb
+   rm ./pdftk.rb
+   ```
+10. Install an assortment of additional packages via `brew install enscript gs imagemagick ruby-build coreutils sqlite parallel tidy-html5`
+11. [Check your rmagick version](#rmagick)
+12. Set up your local MySQL server
+   1. Force link 5.7 version via `brew link mysql@5.7 --force`
+   2. Start mysql with `brew services start mysql`, which uses [Homebrew services](https://github.com/Homebrew/homebrew-services) to manage things for you.
```
Comment (hannahbergam) [inline SETUP.md]: I had to run `brew services start mysql@5.7` here

**Comment by simonguest** [comment]
Looks good to me, especially as we are just updating the SETUP file. Did you also want to include the M1-specific instructions for rbenv?

```export optflags="-Wno-error=implicit-function-declaration"; export LDFLAGS="-L/opt/homebrew/opt/libffi/lib"; export CPPFLAGS="-I/opt/homebrew/opt/libffi/include"; export PKG_CONFIG_PATH="/opt/homebrew/opt/libffi/lib/pkgconfig" | rbenv install 2.5.0```

**Review by hannahbergam — APPROVED** [review]
This is an awesome improvement -  I still hit some hiccups that I commented on but generally I think it's great!

File: SETUP.md (hunk 1 above)
Comment (cat5inthecradle) [reply SETUP.md]: I used sdkman and chose a different JDK provider that was still offering 8. This section probably needs some more work.

File: SETUP.md (hunk 2 above)
Comment (cat5inthecradle) [reply SETUP.md:121]: Was that _without_ switching to a Rosetta terminal or an `arch -x86_64` shell?

File: SETUP.md (hunk 3 above)
Comment (cat5inthecradle) [reply SETUP.md]: yes, good catch. One change from the old docs is that we had a lot of `maybe your version is different` notes with alternate commands. Not sure if I should replicate that, or just assume it's going to get saved as `mysql@5.7`

File: SETUP.md [hunk 4]
```diff
@@ -36,26 +36,89 @@ You can do Code.org development using OSX, Ubuntu, or Windows (running Ubuntu in
...
+11. Install an assortment of additional packages via `brew install enscript gs imagemagick ruby-build coreutils sqlite parallel tidy-html5`
+12. [Check your rmagick version](#rmagick)
+13. Set up your local MySQL server
+   1. Force link 5.7 version via `brew link mysql@5.7 --force`
+   2. Start mysql with `brew services start mysql@5.7`, which uses [Homebrew services](https://github.com/Homebrew/homebrew-services) to manage things for you.
+14. Set up rbenv
+   3.  Run `rbenv init`
+   4.  add `eval "$(rbenv init - zsh)"` to your "~/.zshrc"
+   5.  Run `source ~/.zshrc` to pick up these changes immediately
+15. Install **Ruby 2.5.0**
```
Comment (cat5inthecradle) [inline SETUP.md]: I've got these twice for some reason? Need to revisit this and clean up.

**Pull request merged**